tools/graph_loading/construct_graph.py

```diff
@@ -0,0 +1,680 @@
+                'token_type_ids': th.cat(type_ids, dim=0)}
+
+def parse_tokenize(op):
+    """ Parse the tokenization configuration
+
+    The parser returns a function that tokenizes text with HuggingFace tokenizer.
+    The tokenization function returns a dict of three Pytorch tensors.
+
+    Parameters
+    ----------
+    op : dict
+        The configuration for the operation.
+
+    Returns
+    -------
+    callable : a function to process the data.
+    """
+    tokenizer = BertTokenizer.from_pretrained(op['bert_model'])
+    max_seq_length = int(op['max_seq_length'])
+    return Tokenizer(tokenizer, max_seq_length)
+
+def parse_feat_ops(confs):
+    """ Parse the configurations for processing the features
+
+    The feature transformation:
+    {
+        "feature_col":  ["<column name>", ...],
+        "feature_name": "<feature name>",
+        "data_type":    "<feature data type>",
+        "transform":    {"name": "<operator name>", ...}
+    }
+
+    Parameters
+    ----------
+    confs : list
+        A list of feature transformations.
+
+    Returns
+    -------
+    list of tuple : The operations
+    """
+    ops = []
+    for feat in confs:
+        # TODO(zhengda) we will support data type in the future.
+        dtype = None
+        if 'transform' not in feat:
+            transform = None
+        else:
+            transform = feat['transform']
+            assert 'name' in transform, \
+                    "'name' must be defined in the transformation field."
+            if transform['name'] == 'tokenize_hf':
+                transform = parse_tokenize(transform)
+            else:
+                raise ValueError('Unknown operation: {}'.format(transform['name']))
+        feat_name = feat['feature_name'] if 'feature_name' in feat else None
+        assert 'feature_col' in feat, \
+                "'feature_col' must be defined in a feature field."
+        ops.append((feat['feature_col'], feat_name, dtype, transform))
+    return ops
+
+#################### The main function for processing #################
+
+def process_features(data, ops):
+    """ Process the data with the specified operations.
+
+    This function runs the input operations on the corresponding data
+    and returns the processed results.
+
+    Parameters
+    ----------
+    data : dict
+        The data stored as a dict.
+    ops : list of tuples
+        The operations. Each tuple contains two elements. The first element
+        is the data name and the second element is a Python function
+        to process the data.
+
+    Returns
+    -------
+    dict : the key is the data name, the value is the processed data.
+    """
+    new_data = {}
+    for feat_col, feat_name, dtype, op in ops:
+        # If the transformation is defined on the feature.
+        if op is not None:
+            res = op(data[feat_col])
+            if isinstance(res, dict):
+                for key, val in res.items():
```

Comment:
i think we need to settle how to deal with this case. some preprocessing functions output multiple arrays instead of a single array. how do we assign names to these arrays.

Reply:
define rules for naming multipe output tensors: https://github.com/awslabs/graphstorm/issues/40